app/graphql/mutations/create_cohort.rb

```diff
@@ -23,22 +23,34 @@ class CreateCohort < ApplicationQuery
              }
     argument :ends_at, GraphQL::Types::ISO8601DateTime, required: false
 
-    description 'Create a new cohort'
+    description "Create a new cohort"
 
     field :cohort, Types::CohortType, null: true
 
     def resolve(_params)
-      notify(
-        :success,
-        I18n.t('shared.notifications.done_exclamation'),
-        I18n.t('mutations.create_cohort.success_notification')
-      )
+      cohort = create_cohort
+
+      if cohort.persisted?
+        notify(
+          :success,
+          I18n.t("shared.notifications.done_exclamation"),
+          I18n.t("mutations.create_cohort.success_notification")
+        )
+
+        { cohort: }
+      else
+        notify(
+          :error,
+          I18n.t("shared.notifications.error"),
+          cohort.errors.full_messages.join(", ")
+        )
 
-      { cohort: create_cohort }
+        { cohort: nil }
+      end
     end
 
     def create_cohort
-      course.cohorts.create!(
+      course.cohorts.create(
```

Comment:
Shoudn't we add a validation for checking the name before attempting to save it? 🤔